src/server/game/Battlegrounds/Battleground.cpp

```diff
@@ -1006,6 +1006,10 @@ void Battleground::RemovePlayerAtLeave(Player* player)
     // BG subclass specific code
     RemovePlayer(player);
 
+    // should remove spirit of redemption
+    if (player->HasAuraType(SPELL_AURA_SPIRIT_OF_REDEMPTION))
+        player->RemoveAurasByType(SPELL_AURA_MOD_SHAPESHIFT);
```

Comment:
Why not the specific aura and instead all auras?

Reply:
Not sure why. It's unlikely to me that there would be other shapeshift auras present if the target has spirit of redemption, but this was the previous method used. I found that this pattern is common to remove `SPELL_AURA_SPIRIT_OF_REDEMPTION`. For consistency I'd keep it like this. 

A refactor can be done to replace all these instances with something more efficient in a separate PR.

Reply:
But tbh that is likely not even possible. I was just trying to think of some edge cases